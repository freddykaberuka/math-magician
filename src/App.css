body {
  text-align: center;
}

.main-calculator {
  margin: 10px;
  background-color: #e8e3e3;
  font-size: 35px;
}

.calculator-wrap {
  border: 1px solid black;
  width: 648px;
}

.calculator-result {
  background-color: gray;
  text-align: right;
  padding: 30px;
  color: #fff;
}

.operation-container {
  display: grid;
  grid-template-columns: 9fr 3fr;
}

.calculator-number {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  background-color: #fff;
  cursor: pointer;
}

.calculator-number button {
  height: 50px;
}

.calculator-zero {
  display: grid;
  grid-template-columns: 2fr 1fr;
  background-color: #fff;
  height: 50px;
}

.calculator-operation {
  display: grid;
  grid-template-columns: auto;
}

.calculator-operation button {
  background-color: rgba(232, 110, 10, 0.976);
  height: 50px;
}
