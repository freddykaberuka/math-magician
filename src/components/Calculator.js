import React, { Component } from 'react';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="main-calculator">
        <div className="calculator-wrap">
          <div className="calculator-result">
            0
          </div>
          <div className="operation-container">
            <div className="calculator-upper">
              <div className="calculator-number">
                <button type="button" alt="">AC</button>
                <button type="button" alt="">+/-</button>
                <button type="button" alt="">%</button>
              </div>
              <div className="calculator-lower">
                <div className="calculator-number">
                  <button type="button" alt="">7</button>
                  <button type="button" alt="">8</button>
                  <button type="button" alt="">9</button>
                </div>
                <div className="calculator-number">
                  <button type="button" alt="">4</button>
                  <button type="button" alt="">5</button>
                  <button type="button" alt="">6</button>
                </div>
                <div className="calculator-number">
                  <button type="button" alt="">1</button>
                  <button type="button" alt="">2</button>
                  <button type="button" alt="">3</button>
                </div>
                <div className="calculator-zero">
                  <button type="button" alt="">0</button>
                  <button type="button" alt="">.</button>
                </div>
              </div>
            </div>
            <div className="calculator-operation">
              <button type="button" alt="">รท</button>
              <button type="button" alt="">x</button>
              <button type="button" alt="">-</button>
              <button type="button" alt="">+</button>
              <button type="button" alt="">=</button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default Calculator;
