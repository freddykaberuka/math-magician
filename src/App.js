import React from 'react';
import {
  BrowserRouter as Router, Routes, Route,
} from 'react-router-dom';
import Calculator from './components/Calculator';
import './App.css';
import Home from './pages/Home';
import Quote from './pages/Quote';
import Tabs from './components/Tabs';
import Footer from './components/Footer';

function App() {
  return (
    <>
      <Router>
        <Tabs />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/calculator" element={<Calculator />} />
          <Route path="/quote" element={<Quote />} />
        </Routes>
        <Footer />
      </Router>
    </>
  );
}

export default App;
