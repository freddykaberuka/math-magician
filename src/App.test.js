import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Calculator from './components/Calculator';

describe('App', () => {
  test('renders App Component', () => {
    render(<App />);
  });
});

describe('Test for App component', () => {
  it('It should render calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
