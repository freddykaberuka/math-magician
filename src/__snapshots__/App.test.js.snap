// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for all component It should render calculator component 1`] = `
<div
  className="main-calculator"
>
  <div>
    <p>
      Lets do some math!
    </p>
  </div>
  <div
    className="calculator-wrap"
  >
    <div
      className="calculator-result"
    />
    <div
      className="operation-container"
    >
      <div
        className="calculator-upper"
      >
        <div
          className="calculator-number"
        >
          <button
            alt=""
            onClick={[Function]}
            type="button"
            value="AC"
          >
            AC
          </button>
          <button
            alt=""
            onClick={[Function]}
            type="button"
            value="+"
          >
            +/-
          </button>
          <button
            alt=""
            onClick={[Function]}
            type="button"
            value="%"
          >
            %
          </button>
        </div>
        <div
          className="calculator-lower"
        >
          <div
            className="calculator-number"
          >
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="7"
            >
              7
            </button>
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="8"
            >
              8
            </button>
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="9"
            >
              9
            </button>
          </div>
          <div
            className="calculator-number"
          >
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="4"
            >
              4
            </button>
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="5"
            >
              5
            </button>
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="6"
            >
              6
            </button>
          </div>
          <div
            className="calculator-number"
          >
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="1"
            >
              1
            </button>
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="2"
            >
              2
            </button>
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="3"
            >
              3
            </button>
          </div>
          <div
            className="calculator-zero"
          >
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="0"
            >
              0
            </button>
            <button
              alt=""
              onClick={[Function]}
              type="button"
              value="."
            >
              .
            </button>
          </div>
        </div>
      </div>
      <div
        className="calculator-operation"
      >
        <button
          alt=""
          onClick={[Function]}
          type="button"
          value="รท"
        >
          รท
        </button>
        <button
          alt=""
          onClick={[Function]}
          type="button"
          value="x"
        >
          x
        </button>
        <button
          alt=""
          onClick={[Function]}
          type="button"
          value="-"
        >
          -
        </button>
        <button
          alt=""
          onClick={[Function]}
          type="button"
          value="+"
        >
          +
        </button>
        <button
          alt=""
          onClick={[Function]}
          type="button"
          value="="
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
